### DISCUSSION TITLE
Create a metrics importer for non-Azure cloud platforms

### DISCUSSION THREAD

**Discussion by jmcook1186** [Green-Software-Foundation/hack#30] [Develop an Idea]
## Story
As a user of a cloud platform other than Azure, I want to be able to retrieve usage metrics for my application. Right now, IF only enables me to do this for Azure VMs.

## Rationale
We have implemented an Azure importer in IF, but no such mdoel exists for other cloud providers. Not all cloud providers expose the same metrics as Azure and the available APIs might be quite different. A good project would be to explore what usage metrics are available from other cloud providers and create a model plugin that hits some API and returns usage data.

## Scope
This should be a *model plugin* that conforms to the IF [model plugin interface](https://if.greensoftware.foundation/tutorials/how-to-build-models).

## Implementation guidelines
- This model MUST conform to the IF model plugin interface
- The model MUST return metadata about the application and usage data (e.g. CPU-util, RAM-used, etc) similar to the [azure importer](https://if.greensoftware.foundation/tutorials/how-to-use-azure-vm)
- The model must return data in the expected units (kWh, seconds, GB...)
- The model must accept `duration` in units of seconds an input.
- The model SHOULD be written in Typescript
- The model MUST have unit tests that demonstrate correct execution
- The model MUST have documentation in the form of a README that explains how to use.

## Difficulty
4/5

## Impact
4/5
Many users do not host their applications on Azure, so this will unlock many use cases for IF.

## Examples and resources

[How to use the azure importer](https://if.greensoftware.foundation/tutorials/how-to-use-azure-vm)

**Comment by usmanzafar** [comment]
Hey mate I do like this idea ? Keen to do collab , are you still interested in pursuing  and register this idea ?

**Reply by jmcook1186** [reply]
Hello - I won't personally be participating - I'm on the IF core team and put this idea up as inspiration for others. Encourage you to try to flesh out some more detail (e.g. what non-Azure provider do you want to target, what APIs are available, etc). Then when you are ready you can publish your idea on the issue board.

**Reply by andreisucala** [reply]
Just FYI, the posted links no longer work.

**Reply by NAMRATA-WOKE** [reply]
@jmcook1186 can you take a look?

**Reply by sann3** [reply]
Azure-importer
https://github.com/Green-Software-Foundation/if-unofficial-models/blob/main/src/lib/azure-importer/README.md

**Reply by usmanzafar** [reply]
hey @andreisucala , i reckon this is worth refining, will have a look in coming week. Are you interested and we can both work in defining problem statement further?
